caucsejunseo/다이나믹(dp)/2025.05.05.1로만들기.c

```diff
@@ -0,0 +1,39 @@
+#define _CRT_SECURE_NO_WARNINGS
+#include <stdio.h>
+#include <stdlib.h>
+#define MIN(a,b) ((a) < (b) ? (a) : (b))
+#define MAX 1000001
+
+
+int main() {
+
+    int N = 0;
+    scanf("%d", &N);
+
+    int* dp = (int*)malloc(sizeof(int) * MAX);
```

Comment:
malloc은 동적 메모리 할당에 사용하는 함수인데, MAX 값을 상수로 정해놓고 이를 기반으로 할당하는 것은 사실상 정적 메모리 할당과 다를 바 없어 보이네요..! 정적 메모리 할당으로도 충분히 해결할 수 있는 문제라서 다음과 같이 수정하면 훨씬 간단한 코드가 될 것 같습니다.

```suggestion
    int dp[1000001];
```